src/Aquarius.Client/Samples/Client/SamplesClient.cs

```diff
@@ -23,18 +23,34 @@ public static ISamplesClient CreateConnectedClient(string baseUrl, string apiTok
             return new SamplesClient(baseUrl, apiToken);
         }
 
+        public static ISamplesClient CreateTestClient(IServiceClient client)
+        {
+            return new SamplesClient(client, string.Empty);
+        }
+
         public AquariusServerVersion ServerVersion { get; }
 
-        private JsonServiceClient _client;
+        private IServiceClient _client;
 
         private SamplesClient(string baseUrl, string apiToken)
+            :this(new SdkServiceClient(UriHelper.ResolveEndpoint(baseUrl, "/api", Uri.UriSchemeHttps)), apiToken)
         {
-            _client = new SdkServiceClient(UriHelper.ResolveEndpoint(baseUrl, "/api", Uri.UriSchemeHttps));
-            _client.Headers.Add(AuthorizationHeaderKey, $"token {apiToken}");
+        }
+
+        private SamplesClient(IServiceClient client, string apiToken)
+        {
+            _client = client;
+
+            Client.AddHeader(AuthorizationHeaderKey, $"token {apiToken}");
 
             ServerVersion = GetServerVersion();
 
-            Log.Info($"Connected to {_client.BaseUri} ({ServerVersion}) ...");
+            Log.Info($"Connected to {GetBaseUri()} ({ServerVersion}) ...");
+        }
+
+        private string GetBaseUri()
+        {
+            return (Client as JsonServiceClient)?.BaseUri ?? "http://fake.aqsamples.com/api";
```

Comment:
It might be better to through rather than return a garbage URI.

Also, samples must use HTTPS. I'm not sure if we want to enforce this here though: it could be handy to be able to point this client at a diagnostic tool like fiddler.